test/Schedule.test.ts

```diff
@@ -0,0 +1,223 @@
+
+    await schedule.scheduleCall(ADDRESS.DOT, 0, 300000, 10000, 5, ethers.utils.hexlify(call.data as string));
+
+    let block_hash = await provider.api.query.system.blockHash(current_block_number);
+    const data = await provider.api.derive.tx.events(block_hash);
+    let event = data.events.filter(item => item.event.data.some(data => data.address == ADDRESS.Schedule && data.topics[0] == iface.getEventTopic(iface.getEvent("ScheduledCall"))));
+
+    if (event.length > 0) {
+      let log = {
+        topics: [event[0].event.data[0].topics[0].toString(), event[0].event.data[0].topics[1].toString(), event[0].event.data[0].topics[2].toString()], data: event[0].event.data[0].data.toString()
+      };
+      let decode_log = await iface.parseLog(log);
+      await expect(schedule.cancelCall(ethers.utils.hexlify(decode_log.args.task_id)))
+        .to.emit(schedule, "CanceledCall")
+        .withArgs(await wallet.getAddress(), ethers.utils.hexlify(decode_log.args.task_id));
+    } else {
+      expect(false).to.be.ok;
+    };
+
+    while (current_block_number < target_block_number) {
+      await empty_block(current_block_number);
+      current_block_number = Number(await provider.api.query.system.number());
+    };
+
+    const final_balance = await tokenDOT.balanceOf(await wallet.getAddress());
+
+    expect(final_balance).to.equal(initial_balance);
+  });
+
+  it("emits RescheduledCall event when rescheduling a call and executes it at a new time", async () => {
+    const call = await dex.populateTransaction.swapWithExactSupply([ADDRESS.DOT, ADDRESS.AUSD], 200, 1);
+    const first_target_block_number = Number(await provider.api.query.system.number()) + 5;
+    const second_target_block_number = Number(await provider.api.query.system.number()) + 10;
+    let iface = new ethers.utils.Interface(SCHEDULE_ABI);
+
+    let current_block_number = Number(await provider.api.query.system.number());
+    await schedule.scheduleCall(ADDRESS.DOT, 0, 300000, 10000, 5, ethers.utils.hexlify(call.data as string));
+
+    let block_hash = await provider.api.query.system.blockHash(current_block_number);
+    const data = await provider.api.derive.tx.events(block_hash);
+    let event = data.events.filter(item => item.event.data.some(data => data.address == ADDRESS.Schedule && data.topics[0] == iface.getEventTopic(iface.getEvent("ScheduledCall"))));
+
+    if (event.length > 0) {
+      let log = {
+        topics: [event[0].event.data[0].topics[0].toString(), event[0].event.data[0].topics[1].toString(), event[0].event.data[0].topics[2].toString()], data: event[0].event.data[0].data.toString()
+      };
+      let decode_log = await iface.parseLog(log);
+      await expect(schedule.rescheduleCall(7, ethers.utils.hexlify(decode_log.args.task_id)))
+        .to.emit(schedule, "RescheduledCall")
+        .withArgs(await wallet.getAddress(), ethers.utils.hexlify(decode_log.args.task_id));
+    } else {
+      expect(false).to.be.ok;
+    };
```

Comment:
@xlc @zjb0807 I could use another set of eyes on this one. For some reason, the event `ScheduledCall` is never emitted and without it, I can't test the reschedule call. I went over this setup a couple of times and can't find the issue. This should pass and the setup is done in the same way as in previous examples.

Reply:
I will try

Reply:
I fixed it, you can refer to it to modify others.

Reply:
There was one additional one. I updated it in a same manner you fixed this example. Thank you! 🙏